app/routes/__index/domains/new.tsx

```diff
@@ -45,16 +47,28 @@ export const action = async ({ request }: ActionArgs) => {
   // the domain name (e.g., `foo` in `foo.username.root.com`).
   const { data } = newDnsRecordParams;
 
-  // Create a pending record... for now
-  // This will most likely be replaced by some other logic
-  await addDnsRequest({
-    username: user.username,
-    type: data.type,
-    name: data.name,
-    value: data.value,
-  });
+  try {
+    const recordId = await createUserRecord({
```

Comment:
This complexity should be abstracted away from the rest of the system. You could move this into the addDnsRequest flow creator (not in a worker, in the flow creator itself)